onnx/utils.py

```diff
@@ -21,3 +24,138 @@ def polish_model(model):  # type: (ModelProto) -> ModelProto
     model = onnx.optimizer.optimize(model)
     onnx.checker.check_model(model)
     return model
+
+
+class Extractor:
+    def __init__(self, model):  # type: ignore
+        self.model = onnx.shape_inference.infer_shapes(model)
+        self.graph = self.model.graph
+        self.wmap = self._buildNameDict(self.graph.initializer)
+        self.vimap = self._buildNameDict(self.graph.value_info)
+
+    @staticmethod
+    def _buildNameDict(objs):  # type: ignore
+        return {obj.name: obj for obj in objs}
+
+    def _filterTensors(self, originals, new_names):  # type: ignore
```

Comment:
nit: rename originals to existing_tensors and new_names to something like to_be_filtered or to_be_extracted?